File: problemas/1047/1047.py
hora_inicial, minuto_inicial, hora_final, minuto_final = map(int, input().split())

inicio = hora_inicial *60 + minuto_inicial

fim = hora_final*60 + minuto_final

if(fim <= inicio):
    fim += 24*60
    
duracao = fim - inicio

dura_hr = int(duracao /60)
dura_min = int(duracao%60)

print("O JOGO DUROU {} HORA(S) E {} MINUTO(S)".format(dura_hr,dura_min))

File: problemas/1047/1047.c
#include <stdio.h>

int main() {
    int horaInicial, minutoInicial, horaFinal, minutoFinal;
    int duracaoHoras, duracaoMinutos;

    scanf("%d %d %d %d", &horaInicial, &minutoInicial, &horaFinal, &minutoFinal);

    int inicio = horaInicial * 60 + minutoInicial;
    int fim = horaFinal * 60 + minutoFinal;

    if (fim <= inicio) {
        fim += 24 * 60;
    }

    int duracaoTotal = fim - inicio;
    
    duracaoHoras = duracaoTotal / 60;
    duracaoMinutos = duracaoTotal % 60;

    printf("O JOGO DUROU %d HORA(S) E %d MINUTO(S)\n", duracaoHoras, duracaoMinutos);

    return 0;
}